lib/rubocop/cop/generator.rb

```diff
@@ -116,8 +116,8 @@ def on_send(node)
         '[modify] A configuration for the cop is added into ' \
         '%<configuration_file_path>s.'
 
-      def initialize(name, output: $stdout)
-        @badge = Badge.parse(name)
+      def initialize(name, output: $stdout, badge: Badge.parse(name))
```

Comment:
Is it necessary to change this interface? It seems redundant to make the client prepare a `Badge.parse(name)` object using the `name` argument's value.

Reply:
The idea was to avoid parsing the badge twice(once in the rake task, once in `Generator`). The interface doesn't change unless you choose to pass in a badge, otherwise one will be calculated when initializing `Generator` as before.

Reply:
Yeah, I understand the compatibility of behavior when the `badge` keyword argument is not specified. Based on that, I would like to raise the following concerns:

- The `badge` keyword argument becoming part of the published interface would require guaranteeing its support for clients in the future.
- There is a design concern regarding the redundancy between the first argument `name` and the `name` used in `badge: Badge.new(name)`. While one option might be to temporarily accept both `name` and `badge` as the first argument, I question whether it is necessary for the client to generate a `Badge` instance.
- This `Generator` is not used at runtime.

Considering these points, it is doubtful whether the added complexity is justified by the convenience it provides.

Reply:
@koic Fair points! I have reverted the changes to `Generator.new`.